packages/transfer/src/index.vue

```diff
@@ -0,0 +1,235 @@
+export const RIGHT_CHECK_CHANGE_EVENT = 'right-check-change'
+
+export default defineComponent({
+  name: 'ElTransfer',
+
+  components: {
+    TransferPanel,
+    ElButton,
+  },
+
+  props: {
+    data: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    titles: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    buttonTexts: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    filterPlaceholder: {
+      type: String,
+      default: '',
+    },
+    filterMethod: {
+      type: Function as PropType<(query: string, item: Record<string, any>) => boolean>,
+    },
+    leftDefaultChecked: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    rightDefaultChecked: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    renderContent: {
+      type: Function as PropType<(option) => VNode>,
+    },
+    modelValue: {
+      type: Array,
+      default() {
+        return []
+      },
+    },
+    format: {
+      type: Object,
+      default() {
+        return {}
+      },
+    },
+    filterable: {
+      type: Boolean,
+      default: false,
+    },
+    props: {
+      type: Object,
+      default() {
+        return {
+          label: 'label',
+          key: 'key',
+          disabled: 'disabled',
+        }
+      },
+    },
+    targetOrder: {
+      type: String,
+      default: 'original',
+      validator: (val: string) => {
+        return ['original', 'push', 'unshift'].includes(val)
+      },
+    },
+  },
+
+  emits: [
+    UPDATE_MODEL_EVENT,
+    CHANGE_EVENT,
+    LEFT_CHECK_CHANGE_EVENT,
+    RIGHT_CHECK_CHANGE_EVENT,
+  ],
+
+  setup(props, { emit, slots }) {
+    const initData = reactive({
+      leftChecked: [],
+      rightChecked: [],
+    })
+
+    const {
+      propsKey,
+      sourceData,
+      targetData,
+    } = useComputedData(props)
+
+    const {
+      onSourceCheckedChange,
+      onTargetCheckedChange,
+    } = useCheckedChange(initData, emit)
+
+    const { addToLeft,
+      addToRight,
+    } = useMove(props, initData, propsKey, emit)
+
+    const leftPanel = ref(null)
+    const rightPanel = ref(null)
+
+    const clearQuery = (which: 'left' | 'right') => {
+      if (which === 'left') {
+        leftPanel.value.query = ''
+      } else if (which === 'right') {
+        rightPanel.value.query = ''
+      }
+    }
+
+    const hasButtonTexts = computed(() => props.buttonTexts.length === 2)
+
+    const leftPanelTitle = computed(() => props.titles[0] || t('el.transfer.titles.0'))
+
+    const rightPanelTitle = computed(() => props.titles[1] || t('el.transfer.titles.1'))
+
+    const panelFilterPlaceholder = computed(() => props.filterPlaceholder || t('el.transfer.filterPlaceholder'))
+
+    watch(() => props.modelValue, (val: Key[]) => emit(UPDATE_MODEL_EVENT, val))
+
+    provide('defaultScopedSlots', computed(() => slots.default))
```

Comment:
I would suggest to use an `optionRender` for `renderContent` props.

```suggestion
  const optionRender = computed(() => option => {
     if (props.renderContent) return props.renderContent(h, option)
     if (slots.default) return slots.default(option)
     return // default implementation
  })
```

Reply:
To render vnodes in template, we can use something like

```ts
components: {
  VNodes: ({vnodes}) => vnodes
}
```

And in template

```
<VNodes :vnodes="rendered">
```

Reply:
I just have tried it, but falied. I can print `option` in `optionRender`, but vnodes was not apeared. Did I write something wrong?

![image](https://user-images.githubusercontent.com/33176053/92302053-8c3d0500-ef9b-11ea-8da6-ac9b4f379781.png)

![image](https://user-images.githubusercontent.com/33176053/92302058-92cb7c80-ef9b-11ea-9de6-50d223135e6d.png)

![image](https://user-images.githubusercontent.com/33176053/92302063-9d861180-ef9b-11ea-842c-bf9f840f6388.png)

![image](https://user-images.githubusercontent.com/33176053/92302075-b55d9580-ef9b-11ea-9084-df6e783f7b57.png)

Reply:
try this?

```
  const optionRender = computed(() => {
     if (props.renderContent) return (option) => props.renderContent(h, option)
     if (slots.default) return option => slots.default(option)
     return option => // default implementation
  })
```

Reply:
You can also add some logging in OptionContent's body.

Reply:
I found the reason why the label is not displayed, because I used `:v-nodes=""`, which may be regarded as a drective.

After I changed it, normal render was right, but `slot.default` and `renderContent` both failed. 

The Console's log is in below and you can also use this [test branch](https://github.com/Ryan2128/element-plus/tree/test/transfer) to reproduce it :

![image](https://user-images.githubusercontent.com/33176053/92303274-e642c800-efa5-11ea-9b4b-2ea31db23d75.png)

Reply:
OK, I has fixed it, the reason is the param of slot.